src/tree/index.tsx

```diff
@@ -0,0 +1,422 @@
+import { create, tsx, diffProperty } from '@dojo/framework/core/vdom';
+import { RenderResult } from '@dojo/framework/core/interfaces';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import { createResourceMiddleware } from '@dojo/framework/core/middleware/resources';
+import theme from '@dojo/framework/core/middleware/theme';
+import focus from '@dojo/framework/core/middleware/focus';
+import { flat } from '@dojo/framework/shim/array';
+
+import { Keys } from '../common/util';
+import Icon from '../icon';
+import Checkbox from '../checkbox';
+import { ListItem } from '../list';
+
+import * as css from '../theme/default/tree.m.css';
+import LoadingIndicator from '../loading-indicator';
+
+export interface TreeNodeOption {
+	id: string;
+	parent: string;
+	value: string;
+	hasChildren: boolean;
+}
+
+export interface TreeProperties {
+	checkable?: boolean;
+	selectable?: boolean;
+	checkedNodes?: string[];
+	expandedNodes?: string[];
+	initialChecked?: string[];
+	initialExpanded?: string[];
+	selectedNode?: string;
+	disabledNodes?: string[];
+	parentSelection?: boolean;
+	onSelect?(id: string): void;
+	onCheck?(id: string, checked: boolean): void;
```

Comment:
do you think `onCheck` and `onExpand` might be better reflecting their `checked` and `expanded` counterparts by simply calling the callback with the array of `checked` / `expanded` nodes? This would simplify the callback by providing symmetry with the value and reflect what ant-design does.

Reply:
hm I feel it depends on how the user of this widget wants to structure their code. By passing `checked` and `expanded` the user wouldn't have to have to keep track of these values in the parent widget, they would also have to use `find` if they wanted to find out which node was just checked. Overall I feel we can update this to pass `checked` and `expanded`